protocol/common/message.go

```diff
@@ -758,10 +759,20 @@ func (m *Message) LoadImage() error {
 	return nil
 }
 
+func isValidLinkPreviewThumbnail(thumbnail LinkPreviewThumbnail) bool {
+	return (thumbnail.DataURI == "" && thumbnail.Width == 0 && thumbnail.Height == 0) ||
+		(thumbnail.DataURI != "" && thumbnail.Width > 0 && thumbnail.Height > 0)
+}
+
 func isValidLinkPreviewForProto(preview LinkPreview) bool {
-	return preview.Title != "" && preview.URL != "" &&
```

Comment:
This would be the breaking change mentioned in the description, I think it's worth handling backward compatibility in this case, otherwise clients won't be able to see unfurled links across versions (not sure if you meant to say that you will address backward compatibility in a separate PR).
But it seems we just want to have a case where `Type` is not set, and use the old condition for those, "fingerprinting" the old version. 
We can drop compatibility when enough people are on the compatible build, what do you think?

Reply:
Exactly, I opted for not having a fallback to the `link` type. Now that you mention, I think it's better to go that route. Will do it.

The backward compatibility issue I mentioned in the PR is more related to migrations. Currently it's not possible to easily migrate existing data with just SQL because the column is JSON serialized.

Reply:
Done